stan/math/gpu/kernels/copy_matrix.cl

```diff
@@ -1,25 +1,22 @@
-R"(
-#ifndef A
-#define A(i, j)  A[j * rows + i]
-#endif
+#define STRINGIFY(src) #src
 
-#ifndef B
-#define B(i, j)  B[j * rows + i]
-#endif
+STRINGIFY(
 /**
  * Copy one matrix to another
  * @param[in] A The matrix to copy.
  * @param[out] B The matrix to copy A to.
  * @param rows The number of rows in A.
  * @param cols The number of cols in A.
  *
- * @note Kernel used in math/gpu/matrix_gpu.hpp
+ * @note Kernel used in math/gpu/matrix_gpu.hpp.
+ *  This kernel uses the helper macros available in helpers.cl.
  */
-__kernel void copy(__global double *A, __global double *B, unsigned int rows,
-  unsigned int cols) {
+__kernel void copy(__global read_only double *A, __global write_only double *B,
+	read_only unsigned int rows, read_only unsigned int cols) {
```

Comment:
I think Steve added access qualifiers to all the kernels because we initially didnt have any of them specified. No changes to the semantics are expected here.

Reply:
👍

Reply:
Yes I added these because I figured it would make the code bit more self documenting. Supposedly some compilers can use these for optimizations, but I have not seen anything that explicitly confirms that